<launch>
    <arg name="camera_namespace" default="zivid_camera"/>

    <group ns="$(arg camera_namespace)">
        <node name="zivid_camera" pkg="zivid_camera" type="zivid_camera_node" output="screen">
            <param name="serial_number" type="str" value=""/>
        </node>

        <node name="$(anon dynparam_general)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters capture/general">
            <param name="filters_gaussian_enabled" type="bool" value="true"/>
            <param name="filters_gaussian_sigma" type="double" value="1.0"/>
            <param name="filters_reflection_enabled" type="bool" value="true"/>
        </node>

        <node name="$(anon dynparam_f0)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters capture/frame_0">
            <param name="enabled" type="bool" value="true"/>
            <param name="brightness" type="double" value="1.0"/>
            <param name="exposure_time" type="int" value="40000"/>
            <param name="gain" type="double" value="1.0"/>
            <param name="iris" type="int" value="21"/>
        </node>

        <node name="$(anon dynparam_f1)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters capture/frame_1">
            <param name="enabled" type="bool" value="true"/>
            <param name="brightness" type="double" value="1.0"/>
            <param name="exposure_time" type="int" value="40000"/>
            <param name="gain" type="double" value="1.0"/>
            <param name="iris" type="int" value="28"/>
        </node>

        <node name="$(anon dynparam_f2)" pkg="dynamic_reconfigure" type="dynparam" args="set_from_parameters capture/frame_2">
            <param name="enabled" type="bool" value="true"/>
            <param name="brightness" type="double" value="1.0"/>
            <param name="exposure_time" type="int" value="40000"/>
            <param name="gain" type="double" value="1.0"/>
            <param name="iris" type="int" value="37"/>
        </node>
    </group>
</launch>
